### DISCUSSION TITLE
Automatically use Turborepo when building with Vercel CLI

### DISCUSSION THREAD

**Discussion by IGassmann** [vercel/vercel#8970] [Ideas]
Vercel allows users to build and deploy their applications from GitHub Action (or any other CI environment) thanks to the Vercel CLI's `build` and `deploy` commands. See [relevant guide](https://vercel.com/guides/how-can-i-use-github-actions-with-vercel). This is useful if, for instance, you want to ensure that your test suite has passed before deploying your application.

However, the command `vercel build` doesn't automatically use Turborepo for building the application when there's a `turbo.json` file. It would be nice if it would like it does on Vercel-hosted builds.

I've tested this by running `vercel build` on a non-monorepo Next.js project configured with Turborepo.

**Comment by amyegan** [comment]
Thanks for reaching out @IGassmann! Can you explain a little more about the problem you're seeing? I want to make sure the team has the info they need to address this issue 🙏

**Reply by IGassmann** [reply]
Hey @amyegan!

When I run `vercel build` locally, it runs `next build` instead of `turbo run build`. Since there's a `turbo.json` file on the project, it should run `turbo run build` instead like it does when you build the app on Vercel.

Does that make sense? :)
